src/test/java/org/folio/edge/sip2/api/MainVerticleTests.java

```diff
@@ -79,35 +79,59 @@ public void canStartMainVericleInjectingSip2RequestHandlers(Vertx vertex, VertxT
 
   @Test
   public void cannotCheckoutWithInvalidCommandCode(Vertx vertex, VertxTestContext testContext) {
-    callService("blablabalb", testContext, vertex, result -> {
+    callService("blablabalb\r", testContext, vertex, result -> {
       assertTrue(result.contains("Problems handling the request"));
     });
   }
 
   @Test
   public void canMakeValidSCStatusRequest(Vertx vertex, VertxTestContext testContext) {
-    callService("9900401.00AY1AZFCA5",
-      testContext, vertex, result -> {
-        String expectedPreLocalTime = "98YYNYNN53" + getFormattedDateString();
-        String expectedPostLocalTime = "1.23|AOfs00000010test|AMChalmers|BXYNNNYNYNNNNNNNYN|ANTL01|AFscreenMessages|AGline|\n";
-        String expectedBlankSpaces = "    ";
-
-        assertEquals(result.substring(0, 18), expectedPreLocalTime);
-        assertEquals(result.substring(18, 22), expectedBlankSpaces);
-        assertEquals(result.substring(28), expectedPostLocalTime);
-    });
+    callService("9900401.00AY1AZFCA5\r",
+        testContext, vertex, result -> {
+          validateExpectedACSStatus(result);
+      });
   }
 
   @Test
-  public void canMakeInvalidStatusRequestAndGetExpectedErrorMessage(Vertx vertex, VertxTestContext testContext) {
-    callService("990231.23", testContext, vertex, result -> {
+  public void canMakeInvalidStatusRequestAndGetExpectedErrorMessage(
+      Vertx vertex, VertxTestContext testContext) {
+    callService("990231.23\r", testContext, vertex, result -> {
       assertTrue(result.contains("Problems handling the request"));
     });
   }
 
-  private String getFormattedDateString(){
+  @Test
+  public void canGetCsResendMessageWhenSendingInvalidMessage(
+      Vertx vertx, VertxTestContext testContext) {
+    String scStatusMessage = "9900401.00AY1AZAAAA\r";
+    callService(scStatusMessage, testContext, vertx, result -> {
+      assertEquals("96\r", result);
+    });
+  }
+
+  @Test
+  public void canGetACSStatusMessageWhenSendingValidMessage(
+      Vertx vertx, VertxTestContext testContext) {
+    String scStatusMessage = "9900401.00AY1AZFCA5\r";
+    callService(scStatusMessage, testContext, vertx, result -> {
+      validateExpectedACSStatus(result);
+    });
+  }
+
+  private String getFormattedDateString() {
     String pattern = "YYYYMMdd";
     SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern);
     return simpleDateFormat.format(new Date());
   }
+
+  private void validateExpectedACSStatus(String acsResponse) {
+    String expectedPreLocalTime = "98YYNYNN53" + getFormattedDateString();
+    String expectedPostLocalTime =
+        "1.23|AOfs00000010test|AMChalmers|BXYNNNYNYNNNNNNNYN|ANTL01|AFscreenMessages|AGline|\r";
+    String expectedBlankSpaces = "    ";
+
+    assertEquals(acsResponse.substring(0, 18), expectedPreLocalTime);
```

Comment:
`assertEquals` is `(expected, actual)` which is the opposite of what you have. I noticed that in another test that I was stealing for my test case. My assertion failed and I was a bit confused until I realized the arguments were reversed. You might want to review your other tests to ensure the args are in the proper order.

Reply:
yeah, I screwed this one up.  Will check others.

Reply:
Swapped arguments.